pandas/tests/io/formats/test_to_csv.py

```diff
@@ -534,3 +534,39 @@ def test_to_csv_compression(self, compression_only,
             result = pd.read_csv(path, index_col=0,
                                  compression=read_compression)
             tm.assert_frame_equal(result, df)
+
+    def test_to_csv_compression_dict(self, compression_only):
+        # GH 26023
+        method = compression_only
+        df = DataFrame({"ABC": [1]})
+        filename = "to_csv_compress_as_dict."
+        filename += "gz" if method == "gzip" else method
+        with tm.ensure_clean(filename) as path:
+            df.to_csv(path, compression={"method": method})
+            read_df = pd.read_csv(path, index_col=0)
+            tm.assert_frame_equal(read_df, df)
+
+    def test_to_csv_compression_dict_no_method(self):
+        # GH 26023
+        df = DataFrame({"ABC": [1]})
+        compression = {"some_option": True}
+        with tm.ensure_clean("out.zip") as path, pytest.raises(ValueError):
```

Comment:
Can you match on the expected message with `pytest.raises` here?

Reply:
do the raises as an inner context manager,e .g.
```
with tm.ensure_clearn(...):
     with pytest.raises(...., msg=msg):
```